tests/testthat/test-new-model.R

```diff
@@ -146,3 +146,29 @@ test_that("new_model() supports `.path` containing a period [BBR-NWMD-010]", {
   mod <- new_model(fs::path_ext_remove(temp_ctl), "path with period")
   expect_true(fs::file_exists(temp_yaml))
 })
+
+test_that("new_model ignores file extension: .ctl [BBR-NWMD-011]", {
+  mod_ctl <- new_model(file.path(system.file("model", package = "bbr", mustWork = TRUE), "nonmem", "basic", "1.ctl"),
+                       .description = "original acop model",
+                       .tags = ORIG_TAGS,
+                       .bbi_args = list(overwrite = TRUE, threads = 4)
+  )
+  expect_identical(mod_ctl$absolute_model_path, file.path(system.file("model", package = "bbr", mustWork = TRUE), "nonmem", "basic", "1"))
+  expect_identical(mod_ctl$model_type,"nonmem")
+  expect_identical(mod_ctl$yaml_md5,"6ccf206e167485b5adf29bc135197929")
+})
+
+test_that("new_model ignores file extension: .mod [BBR-NWMD-011]", {
+  fs::file_move(file.path(system.file("model", package = "bbr", mustWork = TRUE),"nonmem","basic","1.ctl"),
+                file.path(system.file("model", package = "bbr", mustWork = TRUE),"nonmem","basic","1.mod"))
+  mod <- new_model(file.path(system.file("model", package = "bbr", mustWork = TRUE), "nonmem", "basic", "1.mod"),
+                   .description = "original acop model",
+                   .tags = ORIG_TAGS,
+                   .bbi_args = list(overwrite = TRUE, threads = 4)
+  )
+  expect_identical(mod$absolute_model_path, file.path(system.file("model", package = "bbr", mustWork = TRUE), "nonmem", "basic", "1"))
+  expect_identical(mod$model_type,"nonmem")
+  expect_identical(mod$yaml_md5,"6ccf206e167485b5adf29bc135197929")
+  on.exit({fs::file_move(file.path(system.file("model", package = "bbr", mustWork = TRUE),"nonmem","basic","1.mod"),
+                         file.path(system.file("model", package = "bbr", mustWork = TRUE),"nonmem","basic","1.ctl"))})
```

Comment:
The `on.exit` should be placed as close as possible to the thing it is cleaning up so that, if any downstream code croaks, the cleanup is still triggered.  Placing it at the end of the function doesn't have the desired effect because the function is about to exit anyway.

Also, I'd suggest assigning these file names to variables since they're repeated across the two `file_move` calls.

And a nit-pick: This

```
file.path(system.file("model", package = "bbr", mustWork = TRUE),
          "nonmem", "basic", "1.ctl")
```

can be simplified to

```
system.file("model", "nonmem", "basic", "1.ctl",
            package = "bbr", mustWork = TRUE)
```

(ditto for several other spots above)